src/uwp/Microsoft.Build.Net.CoreRuntimeTask/ResourceHandlingTask.cs

```diff
@@ -295,61 +408,70 @@ private void WriteResW(Stream stream, string reswFilePath)
-            string resourcesName = "FxResources." + assemblyName + ".SR.resources";
-            int resourceCount = 0;
+            string neutralResourceLanguage;
+            if (String.IsNullOrEmpty(assembly.Culture) && !TryGetNeutralResourcesLanguageAttribute(assembly, out neutralResourceLanguage))
+            {
+                // Must have NeutralResourcesLanguageAttribute
+                // warning MSB3817: The assembly "<FullPath>\ClassLibrary1.dll" does not have a NeutralResourcesLanguageAttribute on it. To be used in an app package, portable libraries must define a NeutralResourcesLanguageAttribute on their main assembly (ie, the one containing code, not a satellite assembly).
+                return null;
+            }
+
+            List<ResWInfo> reswInfoList = new List<ResWInfo>();
+            string frameworkResourcesName = "FxResources." + assembly.Name.Value + ".SR.resources";
 
             foreach (IResourceReference resourceReference in assembly.Resources)
             {
-                resourceCount++;
-                if (!resourceReference.Resource.IsInExternalFile && resourceReference.Name.Value.Equals(resourcesName, StringComparison.OrdinalIgnoreCase))
+                if (!resourceReference.Resource.IsInExternalFile && resourceReference.Name.Value.EndsWith(".resources", StringComparison.OrdinalIgnoreCase))
                 {
                     const int BUFFERSIZE = 4096;
                     byte[] buffer = new byte[BUFFERSIZE];
                     int index = 0;
 
-                    MemoryStream ms = new MemoryStream(BUFFERSIZE);
-
-                    foreach (byte b in resourceReference.Resource.Data)
+                    using (MemoryStream ms = new MemoryStream(BUFFERSIZE))
                     {
-                        if (index == BUFFERSIZE)
+                        foreach (byte b in resourceReference.Resource.Data)
                         {
-                            ms.Write(buffer, 0, BUFFERSIZE);
-                            index = 0;
+                            if (index == BUFFERSIZE)
+                            {
+                                ms.Write(buffer, 0, BUFFERSIZE);
+                                index = 0;
+                            }
+                            buffer[index++] = b;
+                        }
+                        ms.Write(buffer, 0, index);
+                        ms.Seek(0, SeekOrigin.Begin);
+                        
+                        string resourceFileName = resourceReference.Name.Value.Remove(resourceReference.Name.Value.Length - 10);
+                        if (resourceReference.Name.Value.Equals(frameworkResourcesName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            containsFrameworkResources = true;
+                            reswInfoList.Add(ExtractResourcesFromStream(ms, assembly, resourceFileName, true));
+                            return reswInfoList;
```

Comment:
>return reswInfoList; [](start = 28, length = 20)

why returning here? should we continue the loop. just in case the assembly has FX resources and some other resources?

Reply:
In this case I went with the previous logic that in Framework assemblies we only expect one embedded resource and nothing else.

Reply:
This will not work well with the test assemblies. but I am fine with what you have. small comment will be good. thanks for your effort getting this ready quickly.